### PR TITLE
Fix multiple Coverity issues

### PR DESCRIPTION
**Pull request by tszumski**
1. Add initialization of source_begin_iova_map_sz, shm_buf_num
2. Pass dev_port by reference 
3.  Apply lock before accesing fifo.empty()

### CODE REVIEW

File: media-proxy/include/session-mtl.h [hunk 1]
```diff
@@ -60,7 +60,7 @@ class RxSt20MtlSession : public MtlSession
     size_t source_begin_iova_map_sz;
 #endif
 
-    void copy_connection_params(const mcm_conn_param &request, std::string dev_port);
+    void copy_connection_params(const mcm_conn_param &request, std::string &dev_port);
```
Comment (ko80) [inline media-proxy/include/session-mtl.h:63]: C++ reference style: the ampersand sticks to the argument type definition. 
```suggestion
    void copy_connection_params(const mcm_conn_param& request, std::string& dev_port);
```

Consider updating that in all your work.

Comment (tszumski) [reply media-proxy/include/session-mtl.h:63]: Then .clang-format is not configured correctly.
Can we postpone it for later? Fix .clang-format and apply format globally?

Comment (ko80) [reply media-proxy/include/session-mtl.h:63]: Of course, I have no objections.

**Review by ko80 — DISMISSED** [review]

**Review by Sakoram — APPROVED** [review]
Looks good,

only comment: consider using std::lock_guard instead of standard mutex

**Pull request merged**